plugins/inputs/zfs/zfs.go

```diff
@@ -3,32 +3,11 @@ package zfs
 
 import (
 	_ "embed"
-
-	"github.com/influxdata/telegraf"
 )
 
 //go:embed sample.conf
 var sampleConfig string
 
-type Sysctl func(metric string) ([]string, error)
-type Zpool func() ([]string, error)
-type Zdataset func(properties []string) ([]string, error)
-type Uname func() (string, error)
-
-type Zfs struct {
-	KstatPath      string
-	KstatMetrics   []string
-	PoolMetrics    bool
-	DatasetMetrics bool
-	Log            telegraf.Logger `toml:"-"`
-
-	sysctl   Sysctl   //nolint:unused // False positive - this var is used for non-default build tag: freebsd
-	zpool    Zpool    //nolint:unused // False positive - this var is used for non-default build tag: freebsd
-	zdataset Zdataset //nolint:unused // False positive - this var is used for non-default build tag: freebsd
-	uname    Uname    //nolint:unused // False positive - this var is used for non-default build tag: freebsd
-	version  int64    //nolint:unused // False positive - this var is used for non-default build tag: freebsd
-}
-
```

Comment:
My problem with this change is that it will add redundancy for the config parameters i.e. if we add a new flag we now need to remember to all the different OS files. How about moving the "internal" variable to one OS-dependent implementation and have only one flag "nolint" member in the struct?

Reply:
Well I thought about what we did in `systemd_units` but it seems like this looks the same... Anyway, as all other plugins do the same thing, I'm not vetoing this however, I'm questioning the usefulness of the new code. Up to now you could roll out a configuration for ZFS on all systems and would only see a log message. With your change, Telegraf will error out as some platforms now don't have certain parameters defined and are therefore failing at config-parsing state...

Reply:
I see your point. Reverted to have single struct.